libyul/optimiser/ASTWalker.h

```diff
@@ -130,7 +129,7 @@ struct ForEach: Base
 template<typename Node, typename Entry, typename Visitor>
 void forEach(Entry&& _entry, Visitor&& _visitor)
 {
-	detail::ForEach<Node, std::decay_t<Visitor>>{std::forward<Visitor>(_visitor)}(std::forward<Entry>(_entry));
+	detail::ForEach<Node, Visitor&>{_visitor}(std::forward<Entry>(_entry));
```

Comment:
Could actually skip the ref now, since ``detail::ForEach`` makes it a reference anyways... but ref to ref is just ref, so it doesn't matter and maybe clearer to be explicit about it on both ends.